apps/cyberstorm-nextjs/utils/auth.ts

```diff
@@ -0,0 +1,143 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import hkdf from "@panva/hkdf";
+import { EncryptJWT, JWTPayload, jwtDecrypt } from "jose";
+import { v4 as uuid } from "uuid";
+
+interface UserStatusType {
+  username: string;
+  capabilities: string[];
+  subscription: { expires: string | null };
+  rated_packages: string[];
+  teams: string[];
+}
+
+function isUserStatusType(userStatus: unknown): userStatus is UserStatusType {
```

Comment:
- Based on the discord discussion I expected we'd only check some fields here, but being thorough ain't bad (unless changes to e.g. `rated_packages` in the backend breaks this type guard and thus auth check.
- I had a kind of allergic reaction to to the repetition and have a suggestion (see below). It should do the same thing, so the change is only for readability and thus not absolutely necessary, but you could consider it while fixing the unused import error above
- This could be a nice candidate for unit tests, since lots of repetition makes it's easy have a bug in the implementation, and writing the test cases is rather straightforward

```
const isRecord = (obj: unknown): obj is Record<string, unknown> =>
  obj instanceof Object;

const isStringArray = (arr: unknown): arr is string[] =>
  Array.isArray(arr) && arr.every((s) => typeof s === "string");

function isUserStatusType(userStatus: unknown): userStatus is UserStatusType {
  return (
    isRecord(userStatus) &&
    typeof userStatus.username === "string" &&
    isRecord(userStatus.subscription) &&
    (typeof userStatus.subscription.expires === "string" ||
      typeof userStatus.subscription.expires === null) &&
    isStringArray(userStatus.capabilities) &&
    isStringArray(userStatus.rated_packages) &&
    isStringArray(userStatus.teams)
  );
}
```

Reply:
> Based on the discord discussion I expected we'd only check some fields here, but being thorough ain't bad (unless changes to e.g. rated_packages in the backend breaks this type guard and thus auth check.

Hmm, yeah I guess that checking those kind of doesn't make sense, since we'll be fetching those in a separate call, whenever needed?

I'll put your suggestion there, but with the unused stuff left out.